### PR TITLE
feat(interior): batch update interior natives

### PR DESCRIPTION
**Pull request by ook3D**

### CODE REVIEW

**Comment by ook3D** [comment]
@4mmonium hey, would you be so kind as to take a look over these? im pretty confident i did everything correctly

**Comment by 4mmonium** [comment]
> @4mmonium hey, would you be so kind as to take a look over these? im pretty confident i did everything correctly

Hey there, I haven't ignored this, apologies for not giving you a reply earlier. There's quite a lot of changes done in this PR so I have been backlogging it for a bit. Will first need to check if everything is fine within the PR (such as `cs_type` declarations on parameters) and then merge locally to see if the build still compiles. There's often times where changing certain type declarations can break ABI compatibility, so we have to be careful.

Thank you for contributing and for letting me know! 😊

**Comment by ook3D** [comment]
> > @4mmonium hey, would you be so kind as to take a look over these? im pretty confident i did everything correctly
> 
> Hey there, I haven't ignored this, apologies for not giving you a reply earlier. There's quite a lot of changes done in this PR so I have been backlogging it for a bit. Will first need to check if everything is fine within the PR (such as `cs_type` declarations on parameters) and then merge locally to see if the build still compiles. There's often times where changing certain type declarations can break ABI compatibility, so we have to be careful.
> 
> Thank you for contributing and for letting me know! 😊

perfect, i totally understand, thanks!

File: INTERIOR/ActivateInteriorEntitySet.md [hunk 1]
```diff
@@ -10,6 +10,8 @@ void ACTIVATE_INTERIOR_ENTITY_SET(int interior, char* entitySetName);
 ```
 
 ```
+mark the entity set with the given name in this interior as being active.
```
Comment (PsychoShock) [inline INTERIOR/ActivateInteriorEntitySet.md:13]: `M`*

File: INTERIOR/DeactivateInteriorEntitySet.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ aliases: ["0x420BD37289EEE162","_DISABLE_INTERIOR_PROP"]
 // 0x420BD37289EEE162 0xDBA768A1
 void DEACTIVATE_INTERIOR_ENTITY_SET(int interior, char* entitySetName);
 ```
-
+mark the entity set with the given name in this interior as being inactive.
```
Comment (PsychoShock) [inline INTERIOR/DeactivateInteriorEntitySet.md:11]: `M`*

File: INTERIOR/DisableInterior.md [hunk 3]
```diff
@@ -9,6 +9,8 @@ void DISABLE_INTERIOR(int interiorID, BOOL toggle);
 ```
 
 ```
+to turn off an interior & remove all cost associated with it.
```
Comment (PsychoShock) [inline INTERIOR/DisableInterior.md:12]: > To deactivate an interior and eliminate all associated costs.

Would be better.

File: INTERIOR/EnableExteriorCullModelThisFrame.md [hunk 4]
```diff
@@ -10,6 +10,9 @@ void ENABLE_EXTERIOR_CULL_MODEL_THIS_FRAME(Hash mapObjectHash);
 ```
 
 ```
+culls exterior objects from rendering by model name.
+for use in multiplayer apartments which need to cull exterior shell of building etc.
```
Comment (PsychoShock) [inline INTERIOR/EnableExteriorCullModelThisFrame.md:14]: > Filters out exterior objects from rendering based on their model names. Useful for multiplayer apartments requiring the removal of the exterior shell of buildings, etc.

Would be better.

File: INTERIOR/EnableScriptCullModelThisFrame.md [hunk 5]
```diff
@@ -1,13 +1,14 @@
...
 ---
-## _ENABLE_SCRIPT_CULL_MODEL_THIS_FRAME
+## ENABLE_SHADOW_CULL_MODEL_THIS_FRAME
 
 ```c
 // 0x50C375537449F369
-void _ENABLE_SCRIPT_CULL_MODEL_THIS_FRAME(cs_type(Any) Hash mapObjectHash);
+void ENABLE_SHADOW_CULL_MODEL_THIS_FRAME(cs_type(Any) Hash mapObjectHash);
 ```
+culls exterior objects from rendering (cascade shadows only) by model name.
```
Comment (PsychoShock) [inline INTERIOR/EnableScriptCullModelThisFrame.md:11]: > Eliminates exterior objects from rendering, specifically targeting cascade shadows, based on their model names.

Same here

**Review by PsychoShock — CHANGES_REQUESTED** [review]
Hello,

This is a big PR. For nex time, try to limit every PR to 10 natives change so it can be reviewed /  merged faster. I have review probably 15 but took almost 4h, but it's a big work so it's really appreciated by community (like me). I'll continue after this changes are made. 

Thanks

**Comment by ook3D** [comment]
> Hello,
> 
> This is a big PR. For nex time, try to limit every PR to 10 natives change so it can be reviewed / merged faster. I have review probably 15 but took almost 4h, but it's a big work so it's really appreciated by community (like me). I'll continue after this changes are made.
> 
> Thanks

alright sounds good, ill make smaller PRs in the future! 😄

**Comment by freedy69** [comment]
Maybe you could add to the entity set natives that you should call RefreshInterior after you apply changes to the interior for it to show up immediately? Would save some people from frustration im sure (:

**Review by PsychoShock — COMMENTED** [review]
Hey,

Can you add in the format like this:

`**Note:** Add the text there.`

For the RefreshInterior.

Thanks.

**Comment by ook3D** [comment]
> Hey,
> 
> Can you add in the format like this:
> 
> `**Note:** Add the text there.`
> 
> For the RefreshInterior.
> 
> Thanks.

of course, will fix this

**Comment by ook3D** [comment]
any update on this? 😄

**Comment by PsychoShock** [comment]
For a better chance to be PR, you should separate them to 7-8 natives per PR. The issue right now is, if something is broken and need to be revert, it's a __lot__ of natives that will be affected too. It's also easier to review them. Just saying that on personnal experience.

**Comment by ook3D** [comment]
> For a better chance to be PR, you should separate them to 7-8 natives per PR. The issue right now is, if something is broken and need to be revert, it's a **lot** of natives that will be affected too. It's also easier to review them. Just saying that on personnal experience.

fair enough

**Comment by AvarianKnight** [comment]
Thank you for the PR.

We're currently not accepting batch updates to natives.

**Pull request closed without merging**